### PR TITLE
Fix proto `package` and ` java_package` issue

### PR DESCRIPTION
**Pull request by namelessssssssssss**
Fix issue https://github.com/apache/dubbo/issues/13496

* Add provider config `useJavaPackageAsPath`
* `useJavaPackageAsPath=ture` : Provider will export with java_package as path, which will cause exception when `package` and `java_package` are different.
* `useJavaPackageAsPath=false` (default) : Provider will export with package as path.

### CODE REVIEW

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java [hunk 1]
```diff
@@ -548,6 +548,15 @@ private void doExportUrls(RegisterTypeEnum registerType) {
         final boolean serverService = ref instanceof ServerService;
         if (serverService) {
             serviceDescriptor = ((ServerService) ref).getServiceDescriptor();
+            ProviderConfig providerConfig = getScopeModel()
```
Comment (EarthChen) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java:551]: Is it possible to do this in a simpler way?

Comment (chickenlj) [reply dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java:551]: Has the `url` instance already been generated at this point?

**Review by EarthChen — APPROVED** [review]
LGTM

**Review by icodening — APPROVED** [review]
LGTM

**Pull request merged**